src/workerd/api/message-channel.c++

```diff
@@ -0,0 +1,66 @@
+#include "message-channel.h"
+
+#include <workerd/api/events.h>
+#include <workerd/jsg/jsg.h>
+
+namespace workerd::api {
+
+MessagePort::~MessagePort() noexcept(false) {
+  // Technically we should dispatch a close event whenever this object
+  // is garbage collected. Unfortunately, we cannot guarantee that the destructor
+  // will always have the isolate lock.
+  // Let's keep in mind that the destructor might be running during v8 GC
+  // in which you shouldn't execute any javascript at all.
+  // dispatchEvent(js, Event::constructor(kj::str("close"), kj::none));
+}
+
+jsg::Ref<MessagePort> MessagePort::constructor() {
+  return jsg::alloc<MessagePort>();
+}
+
+void MessagePort::disentangle(jsg::Lock &js) {
+  KJ_IF_SOME(e, entangledWith) {
+    // Fire an event named close at otherPort.
+    e->dispatchEvent(js, CloseEvent::constructor());
```

Comment:
In general practice, the static constructor is meant to the be implementation of the `new T` constructor in javascript. For internal construction, use `jsg::alloc<T>` instead.

```suggestion
    e->dispatchEvent(js, jsg::alloc<CloseEvent>());
```